otri/filtering/filters/group_filter.py

```diff
@@ -0,0 +1,160 @@
+from ..filter import Filter, Stream, Sequence, Mapping, Any
+from datetime import timedelta, datetime, timezone
+from ...utils import time_handler as th
+
+
+class GroupFilter(Filter):
+    '''
+    Groups price atoms to a lower resolution.\n
+
+    Inputs:
+        Oredered by datetime atoms.\n
+    Outputs:
+        Atoms with lower resolution\n
+    '''
+
+    def __init__(self, inputs: str, outputs: str, resolution: timedelta = timedelta(days=1), datetime_key: str = "datetime", volume_key: str = None):
+        '''
+        Parameters:\n
+             input, output : Sequence[str]
+                Name for input/output streams.\n
+            resolution : timedelta
+                Target resolution, must be lower than current resolution. Valid values are: 1,2,3,4,5,6,8,10,12,15,30 seconds or minutes 1,2,3,4,6,8,12 hours 1 day.\n
+            datetime_key : str
+                Key name for the datetime value.\n
+            volume_key : str
+                Key name for the volume value. If None it won't group volume value.\n
+        '''
+        super().__init__(
+            inputs=[inputs],
+            outputs=[outputs],
+            input_count=1,
+            output_count=1
+        )
+        self.__resolution = resolution
+        self.__number_res = self._timedelta_to_number(resolution)
+        self.__datetime_key = datetime_key
+        self.__volume_key = volume_key
+
+    def setup(self, inputs: Sequence[Stream], outputs: Sequence[Stream], state: Mapping[str, Any]):
+        '''
+        Used to save references to streams and reset variables.\n
+        Called once before the start of the execution in FilterNet.\n
+
+        Parameters:\n
+            inputs, outputs : Sequence[Stream]
+                Ordered sequence containing the required input/output streams gained from the FilterNet.\n
+            state : Mapping[str, Any]
+                Dictionary containing states to output.\n
+        '''
+        # Call superclass setup
+        super().setup(inputs, outputs, state)
+        self.__cur_dt = datetime(1980, 1, 1, 1, 1)
+        self.__buffer = None
+        self.__cur_high = float("-inf")
```

Comment:
I wouldn't just assume that atoms that pass through this Filter will have these exact fields, and only these.
I would either isolate all of this in some external function or explicitly say what the schema of input atoms should be.

Reply:
Generalized the filter, now can work for any kind of group-able data